mypy/checker.py

```diff
@@ -4609,34 +4610,28 @@ def has_no_custom_eq_checks(t: Type) -> bool:
                 elif operator in {'in', 'not in'}:
                     assert len(expr_indices) == 2
                     left_index, right_index = expr_indices
-                    if left_index not in narrowable_operand_index_to_hash:
-                        continue
 
-                    item_type = operand_types[left_index]
-                    collection_type = operand_types[right_index]
+                    left_is_narrowable = left_index in narrowable_operand_index_to_hash
+                    right_is_narrowable = right_index in narrowable_operand_index_to_hash
 
-                    # We only try and narrow away 'None' for now
-                    if not is_optional(item_type):
-                        continue
+                    left_type = operand_types[left_index]
+                    right_type = operand_types[right_index]
 
-                    collection_item_type = get_proper_type(builtin_item_type(collection_type))
-                    if collection_item_type is None or is_optional(collection_item_type):
-                        continue
-                    if (isinstance(collection_item_type, Instance)
-                            and collection_item_type.type.fullname == 'builtins.object'):
-                        continue
-                    if is_overlapping_erased_types(item_type, collection_item_type):
-                        if_map, else_map = {operands[left_index]: remove_optional(item_type)}, {}
-                    else:
-                        continue
-                else:
-                    if_map = {}
-                    else_map = {}
+                    if left_is_narrowable:
+                        narrowed_left_type = self.refine_optional_in(left_type, right_type)
+                        if narrowed_left_type:
+                            if_map = {operands[left_index]: narrowed_left_type}
+
+                    elif right_is_narrowable:
```

Comment:
If `left_is_narrowable`, shouldn't we still try to narrow the right side too (if `right_is_narrowable`)?

In particular should this just be `if` rather than `elif`?